src/utils/paths.cpp

```diff
@@ -41,20 +43,20 @@ std::filesystem::path Paths::getHomePath() {
         "https://github.com/JoseExposito/touchegg/issues"};
   }
 
-  const char *workingDir = userInfo->pw_dir;
-  if (workingDir == nullptr) {
+  std::filesystem::path workingDir = std::filesystem::path{userInfo->pw_dir};
+  if (!std::filesystem::exists(workingDir)) {
     throw std::runtime_error{
         "Error getting your home directory path (pw_dir).\n"
         "Please file a bug report at "
         "https://github.com/JoseExposito/touchegg/issues"};
   }
 
-  return std::filesystem::path{workingDir};
+  return std::filesystem::path{workingDir / ".config"};
 }
 
 std::filesystem::path Paths::getUserConfigDirPath() {
   std::filesystem::path homePath = Paths::getHomePath();
-  return std::filesystem::path{homePath / ".config" / "touchegg"};
+  return std::filesystem::path{homePath / "touchegg"};
```

Comment:
I think that it'd be better to leave `getHomePath` as it is and inside this method, follow the spec steps:

- Check the value of `$XDG_CONFIG_HOME`
- Does it exist? 
  -  Yes: Use it
  - No: Fallback to `homePath / .config`